### PR TITLE
[Github] Add support for building libc docs in Github actions

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch adds support for building the libc docs in Github actions. This eanbles easily diagnosing doc build failures/warnings in PRs and at the tip of tree.

### CODE REVIEW

**Review by tstellar — APPROVED** [review]
LGTM.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -69,4 +73,10 @@ jobs:
         run: |
           cmake -B clang-build -GNinja -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="clang" -DLLVM_ENABLE_SPHINX=ON ./llvm
           TZ=UTC ninja -C clang-build docs-clang-html docs-clang-man
+      - name: Build libc docs
+        if: steps.docs-changed-subprojects.outputs.libc_any_changed == 'true'
+        run: |
+          cmake -B libc-build -GNinja -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_RUNTIMES="libc" -DLLVM_ENABLE_SPHINX=ON ./runtimes
```
Comment (michaelrj-google) [inline .github/workflows/docs.yml:79]: this also needs `-DLIBC_INCLUDE_DOCS=ON`

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -69,4 +73,10 @@ jobs:
...
+          TZ=UTC ninja -C libc-build docs-libc-html
```
Comment (michaelrj-google) [inline .github/workflows/docs.yml:80]: `libc-build` isn't the target for building LLVM libc. The target is just `libc`.

**Review by michaelrj-google — COMMENTED** [review]
Overall good, but there are some changes needed.

File: .github/workflows/docs.yml (hunk 2 above)
Comment (boomanaiden154) [reply .github/workflows/docs.yml]: Not sure how the target ended up in there. We only need the `docs-libc-html` target.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (boomanaiden154) [reply .github/workflows/docs.yml:79]: This doesn't seem necessary to set explicitly? It gets its initial value from `LLVM_INCLUDE_DOCS` which is set to `ON` by default.

Happy to add the flag if so desired though.

Comment (michaelrj-google) [reply .github/workflows/docs.yml:79]: ah, you're right. No need to add the extra flag.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Pull request merged**